Block/Configuration.php

```diff
@@ -92,6 +94,11 @@ public function getConfiguration()
             }
         }
 
+        $productId = null;
+        if ($config->isClickConversionAnalyticsEnabled() && $request->getControllerName() === 'product') {
```

Comment:
Maybe checking also the action ? (if the goal is only to get the ProductId on product view ?)